app/controllers/concerns/foreman/controller/authentication.rb

```diff
@@ -71,14 +71,17 @@ def sso_authentication
 
   def set_current_user(user)
     User.current = user
-
     # API access resets the whole session and marks the session as initialized from API
     # such sessions aren't checked for CSRF
     # UI access resets only session ID
     if api_request?
+      jwt_data = { sso_method: session[:sso_method], expires_at: session[:expires_at] }
```

Comment:
I think the better approach would be to make [backup_session_content](https://github.com/theforeman/foreman/blob/ae96cb70130968bdcc8b7a9a9af7edc7c6e0d9a8/app/controllers/concerns/foreman/controller/session.rb#L17-L21) accept the keys you want to keep (`sso_method`, `expires_at`) so you can call it like this `backup_session_content([:sso_method, :expires_at]) { reset_session }`.

This way this generic auth concern does not need to know anything about JWTs.

Reply:
I updated the backup_session_content method to accept keys.
However, I still had to keep the condition for checking "SSO::OpenidConnect" in this method. I have added comments over the method for an explanation. Please let me know thoughts for it.